.github/workflows/clang-tidy.yaml

```diff
@@ -0,0 +1,74 @@
+name: clang-tidy
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  clang-tidy:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Environment
+        env:
+          PROTOBUF_VERSION: '23.3'
+          ABSEIL_CPP_VERSION: '20230125.3'
+          CXX_STANDARD: '14'
+        run: |
+          sudo apt update -y
+          sudo apt install -y --no-install-recommends --no-install-suggests \
+          build-essential
+          sudo -E ./ci/setup_cmake.sh
+          sudo -E ./ci/setup_ci_environment.sh
+          sudo -E ./ci/setup_googletest.sh
+          sudo -E ./ci/install_abseil.sh
+          sudo -E ./ci/install_protobuf.sh
+          sudo -E ./ci/setup_grpc.sh -m -p protobuf -p abseil-cpp
```

Comment:
This will install dependencies from source, rebuilding a lot of code.

For clang-tidy, using the most up to date versions of dependencies is not relevant.

Better install packages that come with the platform, see apt install from the iwyu build.

Reply:
@marcalff  Is there any way to install `grpc ` from package managers? I don't see any. Please let me know if there are any. Because building `gRPC` from source code takes lot of time.

Reply:
Let's put this aside from now, the same issue exists with the maintainers build for warnings, and iwyu.